src/services/catalog.service.ts

```diff
@@ -0,0 +1,198 @@
+import * as angular from 'angular';
+import * as _ from 'lodash';
+
+export class CatalogService {
+  static $inject = ['Constants', '$filter'];
+
+  public $filter: any;
+  private constants: any;
+  private categories: any;
+
+  constructor(constants: any, $filter: any, $scope: any) {
+    this.constants = constants;
+    this.$filter = $filter;
+    this.categories = this.constants.SERVICE_CATALOG_CATEGORIES;
+  }
+
+  public getServiceItem(resource: any) {
+    return new ServiceItem(resource, this);
+  }
+
+  public getImageItem(resource: any) {
+    return new ImageItem(resource, this);
+  }
+
+  public getCategoryFromSubCategory(subCategory: string) {
```

Comment:
I had several comments about these methods against the other file, i think they all still apply here if you can just look back at those in the history.